tools/releaseTools.psm1

```diff
@@ -246,32 +251,148 @@ function Get-NewOfficalPackage
+                    Add-Member -Type ScriptProperty -Name Published -Value { $this.Metadata['published']} -PassThru |
+                        Where-Object { Test-IncludePackageVersion -NewVersion $_.Version -Version $package.version}
+
+                $revsionRegEx = Get-MatchingMajorMinorRegEx -Version $package.version
+                $newPackage = $versions |
+                    Sort-Object -Descending |
+                        Select-Object -First 1
+
+                # Get the newest matching revision
+                $newRevision = $versions |
+                    Where-Object {$_.Version -match $revsionRegEx } |
+                        Sort-Object -Descending |
+                            Select-Object -First 1
 
                 # If the current package has a different version from the version in the csproj, print the details
-                if($newPackage -and $newPackage.Version.ToString() -ne $package.version)
+                if ($newRevision -and $newRevision.Version.ToString() -ne $package.version -or $newPackage -and $newPackage.Version.ToString() -ne $package.version -or $IncludeAll.IsPresent)
                 {
+                    if ($newRevision)
+                    {
+                        $newRevisionString = $newRevision.Version
+                    }
+                    else
+                    {
+                        # We don't have a new Revision, report the current version
+                        $newRevisionString = $package.Version
+                    }
+
+                    if ($newPackage)
+                    {
+                        $newVersionString = $newPackage.Version
+                    }
+                    else
+                    {
+                        # We don't have a new Version, report the current version
+                        $newVersionString = $package.Version
+                    }
+
                     [pscustomobject]@{
-                        Csproj = $file
+                        Csproj = (Split-Path -Path $file -Leaf)
                         PackageName = $name
                         CsProjVersion = $Package.Version
-                        NuGetVersion = $newPackage.Version
+                        NuGetRevision = $newRevisionString
+                        NuGetVersion = $newVersionString
                     }
                 }
             }
         }
     }
 }
 
+##############################
+#.SYNOPSIS
+# Returns True if NewVersion is newer than Version
+#
+#.PARAMETER Version
+# The current Version
+#
+#.PARAMETER NewVersion
+# The potention replacement version
+#
+#.OUTPUTS
+# True if NewVersion should be considere as a replacement
+##############################
+function Test-IncludePackageVersion
+{
+    param(
+        [string]
+        $NewVersion,
+        [string]
+        $Version
+    )
+
+    $simpleCompare = $Version -notlike '*-*'
```

Comment:
Should we use semantic version instead, would make the checks much simpler. 
```PowerShell
$NewVersion = [System.Management.Automation.SemanticVersion]"6.1.0-preview.4"
$Version = [System.Management.Automation.SemanticVersion]"6.0.4"

if($NewVersion.PreReleaseLabel) { return false } else { return $NewVersion -gt $Version } 
…..

```

Reply:
I had an offline discussion with @adityapatwardhan .  I'll update the synopsis on why the current logic is correct.